packages/account/src/providers/transaction-request/transaction-request.ts

```diff
@@ -424,7 +427,7 @@ export abstract class BaseTransactionRequest implements BaseTransactionRequestLi
     this.pushInput(input);
 
     // Insert a ChangeOutput if it does not exist
-    this.addChangeOutput(recipient, assetId);
+    this.addChangeOutput(recipient, this.baseAssetId);
```

Comment:
```suggestion
    this.addChangeOutput(recipient, message.assetId);
```

All resources are fetched from `Provider.getResourcesToSpend`, which is the graphQL query `coinsToSpend`. 

When the user owns messages that have not been spent yet, they will be returned with the `assetId` property (base asset) already [set to it](https://github.com/FuelLabs/fuels-ts/blob/cb2da0408e38b4af9a1198fa4f3b504929274b7f/packages/account/src/providers/provider.ts#L1087-L1095).

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1976/commits/1239a4bbaac8a320ced1c66eeab51224269e0d8d